server/src/main/java/org/apache/uniffle/server/ShuffleTaskManager.java

```diff
@@ -579,4 +602,11 @@ void removeShuffleDataSync(String appId, int shuffleId) {
   public ShuffleDataDistributionType getDataDistributionType(String appId) {
     return shuffleTaskInfos.get(appId).getDataDistType();
   }
+
+  private void triggerFlush() {
+    synchronized (this.shuffleBufferManager) {
```

Comment:
Sorry, I can't get this point of this change. By the way, it we hold the lock of `ShuffleBufferManager`, do it effect the critical rpc request when sending shuffle-data from client. refer to: https://github.com/apache/incubator-uniffle/blob/53212923328194fafa5ae94aa216ab04f06fbd13/server/src/main/java/org/apache/uniffle/server/buffer/ShuffleBufferManager.java#L120-L125.

Although I think this part is unreasonable because it will concurrent requests reshape to serial. Maybe this could be solved in another PR. But for current codebase, the `triggerFlush` will effect performance

Reply:
> In current codebase, the triggerFlush is only called per rss.server.shuffleBufferManager.trigger.flush.interval, which should be 2-5 mins on server side. That shouldn't be too much overhead?

Yes, it wont.

> As for why triggerFlush is introduced in this PR, its main reason is to reduce UT changes for the behavior changes as preAllocatedSize is decreased in Grpc layer, rather than the cache data layer.

Got it.

> And also, I thought it would be better if we decoupled the flush logic with cacheShuffleData, which causes flush spikes, and frequent full gc.

+1.